packages/fuels-core/src/tokenizer.rs

```diff
@@ -0,0 +1,866 @@
+use crate::Token;
+use fuels_types::errors::Error;
+use fuels_types::param_types::ParamType;
+use fuels_types::utils::has_array_format;
+use hex::FromHex;
+
+#[derive(Default)]
+pub struct Tokenizer {}
+
+impl Tokenizer {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl Tokenizer {
+    /// Takes a ParamType and a value string and joins them as a single
+    /// Token that holds the value within it. This Token is used
+    /// in the encoding process.
+    pub fn tokenize(&self, param: &ParamType, value: String) -> Result<Token, Error> {
```

Comment:
Since it doesn't have any state, we can go for associated functions rather than methods.

i.e.
```rust
let result = Tokenizer::tokenize(param_type, value);
```